### PR TITLE
Enable more clang-tidy checks

### PR DESCRIPTION
**Pull request by masterleinad**
These are the last `modernize-` checks (besides `modernize-loop-convert`) I would like to enable.
All the others didn't detect anything or are controversial. However, I think about disabling the controversial ones instead of enabling the ones we have currently after all the changes with respect to the current checks enabled are in.

### CODE REVIEW

**Comment by masterleinad** [comment]
Since there were no comments so far, I inverted the way of enabling `clang-tidy` `modernize` tests here in the last commit. Some comments regarding the disabled checks:
-  `modernize-pass-by-value` changes a lot of interfaces to pass by value instead of `const &`. Hence, it goes against our philosophy of marking parameters that are not changed inside a function as `const` (and not exposing implementation details).
- `modernize-use-auto` adds `auto` in a lot of places which is always a bit controversial. Sometimes, it is good to know what the actual type is.
- `modernize-use-override` would remove all `virtual` specifiers from all functions marked `override`. We decided to have both specifiers.
- `modernize-use-default-member-init` is controversial. So far, we decided not to use default member initialization.
- `modernize-use-transparent-functors` needs `C++14` enabled.

**Comment by drwells** [comment]
That sounds reasonable: I would prefer to avoid `modernize-use-default-member-init` because of the massive amount of churn it will produce.

File: source/base/data_out_base.cc [hunk 1]
```diff
@@ -3196,14 +3196,14 @@ namespace DataOutBase
       }
 
 
-    out << "attribute \"element type\" string \"";
+    out << R"(attribute "element type" string ")";
```
[the remark this answers is not in the document]
Comment (drwells) [reply source/base/data_out_base.cc]: I also prefer the escaped version: I have trouble remembering that the surrounding parentheses are special.

[the remark this answers is not in the document]
Comment (masterleinad) [reply source/base/data_out_base.cc]: Then I just drop the commit and remove the check. I don't have a particular preference but like to discuss new possibilities.

[the remark this answers is not in the document]
Comment (masterleinad) [reply source/base/data_out_base.cc]: Here you go!

File: include/deal.II/lac/solver_selector.h [hunk 2]
```diff
@@ -110,7 +110,7 @@ class SolverSelector : public Subscriptor
   /**
    * Destructor
    */
-  ~SolverSelector() override;
+  virtual ~SolverSelector() override = default;
```
Comment (bangerth) [inline include/deal.II/lac/solver_selector.h]: good catch!

File: include/deal.II/particles/particle_handler.h [hunk 3]
```diff
@@ -85,7 +85,7 @@ namespace Particles
     /**
      * Destructor.
      */
-    ~ParticleHandler() override;
+    virtual ~ParticleHandler() override = default;
```
Comment (bangerth) [inline include/deal.II/particles/particle_handler.h]: same here!

**Pull request merged**